src/test/java/com/ibm/cloud/sdk/core/test/service/ErrorResponseTest.java

```diff
@@ -73,25 +74,74 @@ public void setUp() throws Exception {
     service.setServiceUrl(getMockWebServerUrl());
   }
 
+  private void verifyException(ServiceResponseException e, int expectedStatusCode, String expectedMessage,
+      String expectedResponseBody) {
+
+    assertEquals(e.getStatusCode(), expectedStatusCode);
+    assertEquals(e.getMessage(), expectedMessage);
+    assertEquals(e.getResponseBody(), expectedResponseBody);
+  }
+
   /**
    * Test HTTP status code 400 (Bad Request) error response.
    */
   @Test
   public void testBadRequest() {
 
     String message = "The request failed because the moon is full.";
+    String responseBody = "{\"error\": \"" + message + "\"}";
     server.enqueue(new MockResponse()
         .setResponseCode(400)
         .addHeader(CONTENT_TYPE, HttpMediaType.APPLICATION_JSON)
-        .setBody("{\"error\": \"" + message + "\"}"));
+        .setBody(responseBody));
+
+    try {
+      service.testMethod().execute();
+    } catch (Exception e) {
+      assertTrue(e instanceof BadRequestException);
+      BadRequestException ex = (BadRequestException) e;
+      verifyException(ex, 400, message, responseBody);
+    }
+  }
+
+  /**
+   * Test HTTP status code 400 (Bad Request) error response with no response body.
+   */
+  @Test
+  public void testBadRequestNoBody() {
+
+    String message = "The request failed because the moon is full.";
+    server.enqueue(new MockResponse()
+        .setResponseCode(400)
+        .addHeader(CONTENT_TYPE, HttpMediaType.APPLICATION_JSON));
+
+    try {
+      service.testMethod().execute();
+    } catch (Exception e) {
+      assertTrue(e instanceof BadRequestException);
+      BadRequestException ex = (BadRequestException) e;
+      verifyException(ex, 400, "Bad request", null);
+    }
+  }
+
+  /**
+   * Test HTTP status code 400 (Bad Request) error response with non-JSON response body.
+   */
+  @Test
+  public void testBadRequestTextBody() {
```

Comment:
I was equally surprised to find that we didn't have a test for a non-JSON error response body.